packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,429 @@
+   * 
+   *    For the G.1X worker type, each worker maps to 1 DPU (4 vCPU, 16 GB of memory,
+   *    64 GB disk), and provides 1 executor per worker. We recommend this worker type
+   *    for memory-intensive jobs.
+   * 
+   *    For the G.2X worker type, each worker maps to 2 DPU (8 vCPU, 32 GB of memory,
+   *    128 GB disk), and provides 1 executor per worker. We recommend this worker type
+   *    for memory-intensive jobs.
+   */
+  public readonly workerType?: WorkerType;
+  
+  /**
+   * The job name
+   */
+  public readonly jobName: string;
+
+  constructor(scope: Construct, id: string, props: ConnectionInputProps) {
+    super(scope, id);
+
+    this.allocatedCapacity = props.allocatedCapacity;
+    this.command = props.command;
+
+    if (props.connections) {
+      props.connections.forEach(c => this.addConnection(c));
+    }
+
+    this.defaultArguments = props.defaultArguments;
+    this.description = props.description;
+    this.maxConcurrentRuns = props.maxConcurrentRuns;
+    this.glueVersion = props.glueVersion;
+    this.logUri = props.logUri;
+    this.maxCapacity = props.maxCapacity;
+    this.maxRetries = props.maxRetries;
+    this.name = props.name;
+    this.notifyDelayAfter = props.notifyDelayAfter;
+    this.numberOfWorkers = props.numberOfWorkers;
+
+    // TODO: add role
+    if (!(props.role)) {
+      this.role = new iam.Role(this, 'JobRole')
+    }
+
+    this.securityConfiguration = props.securityConfiguration;
+    this.tags = props.tags;
+    this.timeout = props.timeout;
+    this.workerType = props.workerType;
+
+    const jobResource = new CfnJob(this, 'Job', {
+      allocatedCapacity: this.allocatedCapacity,
+      command: this.command,
+      connections: Lazy.anyValue({ produce: () => this.connections }, { omitEmptyArray: true }),
+      defaultArguments: JSON.stringify(this.defaultArguments),
+      description: this.description,
+      
+      executionProperty: {
+        maxConcurrentRuns: this.maxConcurrentRuns
+      },
+
+      glueVersion: this.glueVersion,
+      logUri: this.logUri,
+      maxCapacity: this.maxCapacity,
+      maxRetries: this.maxRetries,
+      name: this.name,
+
+      notificationProperty: {
+        notifyDelayAfter: this.notifyDelayAfter
+      },
+
+      numberOfWorkers: this.numberOfWorkers,
+
+      role: this.role,
+      securityConfiguration: this.securityConfiguration,
+      tags: this.tags,
+      timeout: this.timeout.toMinutes(),
+      workerType: this.workerType
+    })
+
+    this.jobName = this.getResourceNameAttribute(jobResource.ref);
+
+    // TODO
+    // this.jobArn = this.stack.formatArn({
+    //   service: 'glue',
+    //   resource: 'job',
+    //   // resourceName: `${this.database.databaseName}/${this.tableName}`
+    // });
+
+    this.node.defaultChild = jobResource
+  }
+
+
+  /**
+   * Adds a connection to the glue job.
+   */
+  public addConnection(connection: Connection) {
+    this.connections.push(connection);
+  }
+}
```

Comment:
Pretty much the same comments here as in the `connection.ts` file:

1. `@attribute` in the wrong places.
2. Wrong ID of the child L1 (should be 'Resource').
3. Creation properties are 1:1 with their CloudFormation equivalents - they should be flattened instead.
4. Too much exposed properties on the `IJob` interface.
5. Too many `object` properties.